src/cubic-b-spline.cc

```diff
@@ -215,6 +215,10 @@ namespace trajectory
 
     typedef boost::numeric::converter<size_type, double> Double2SizeType;
 
+    if (t == knots_[static_cast<size_t>(imax+1)]) //We are exactly at the end of the spline
+    {
+      return imax;
+    }
```

Comment:
Ok.
It does not concern the uniform case. All knots-defined splines (Splines one creates with a knot vector) are concerned, even the uniform ones, but I think it indeed should be placed in the dichotomy block, as it is basically here to avoid the endless case at the end of it.